src/main/scala/is/hail/stats/LinearRegressionCombiner.scala

```diff
@@ -0,0 +1,144 @@
+package is.hail.stats
+
+import breeze.linalg.{DenseMatrix, DenseVector, diag, dim, inv}
+import breeze.numerics._
+import is.hail.annotations.{Annotation, Region, RegionValueBuilder}
+import is.hail.expr.types._
+import net.sourceforge.jdistlib.T
+
+object LinearRegressionCombiner {
+  val typ: Type = TStruct(
+    "beta" -> TArray(TFloat64()),
+    "standard_error" -> TArray(TFloat64()),
+    "p_value" -> TArray(TFloat64()),
+    "n" -> TInt64())
+
+  val xsType = TArray(TFloat64())
+}
+
+class LinearRegressionCombiner(nxs: Int) extends Serializable {
+  var n = 0L
+  var xs = DenseVector.zeros[Double](nxs)
+  var xtx = DenseMatrix.zeros[Double](nxs, nxs)
+  var xty = DenseVector.zeros[Double](nxs)
+  var yty = 0.0
+
+  val xsType = LinearRegressionCombiner.xsType
+
+  def merge(y: Double, xsArray: IndexedSeq[Double]) {
+    assert(nxs == xsArray.length)
+    xs = DenseVector(xsArray.toArray)
+
+    n += 1
+    xtx :+= xs * xs.t
+    xty :+= xs * y
+    yty += y * y
+  }
+
+  def merge(region: Region, y: Double, xsOffset: Long) {
+    val length = xsType.loadLength(region, xsOffset)
+    assert(nxs == length)
+
+    var i = 0
+    while (i < length) {
+      if (xsType.isElementMissing(region, xsOffset, i))
+        return
+
+      xs(i) = region.loadDouble(xsType.loadElement(region, xsOffset, i))
+      i += 1
+    }
+
+    n += 1
+    xtx :+= xs * xs.t
+    xty :+= xs * y
+    yty += y * y
+  }
+
+  def merge(other: LinearRegressionCombiner) {
+    n += other.n
+    xtx :+= other.xtx
+    xty :+= other.xty
+    yty += other.yty
+  }
+
+  def computeResult(): (DenseVector[Double], DenseVector[Double], DenseVector[Double]) = {
+    val b = xtx \ xty
+    val rse2 = 1.0 / (n - nxs) * (yty - (xty dot b)) // residual standard error squared
+    val se = sqrt(rse2 * diag(inv(xtx)))
+    val t = b /:/ se
+    (b, se, t)
+  }
+
+  def result(rvb: RegionValueBuilder) {
+    val (b, se, t) = computeResult()
+
+    rvb.startStruct()
+
+    if (n != 0) {
```

Comment:
You ought to check that `n > nxs` before computing the result. otherwise you don't have a positive number of degrees of freedom and nothing makes sense. In `linear_regression` we fatal if this isn't true, but there it's also the case that `n` is the same for all rows because we mean impute. Here I think missing is fine, and the reason for missing will be clear since `n` is returned in the result.

And I think you should pull the `if` outside rather than having four `if` of the same condition.